Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -0,0 +1,514 @@
+                            basePath="/senders"
+                            source="id"
+                            target="flow_id"
+                            reference="senders"
+                            link="show"
+                        >
+                            <div
+                                style={{
+                                    margin: 5,
+                                    padding: 10,
+                                }}
+                            >
+                                <SingleFieldList linkType="show">
+                                    <ChipConditionalLabel source="label" />
+                                </SingleFieldList>
+                            </div>
+                        </ReferenceManyField>
+                    </SingleFieldList>
+                </ReferenceManyField>
+            </div>
+        </>
+    );
+};
+
+const OutputSourceAssociation = ({ outputs, isExpend }) =>
+    outputs.map(([output_name, output_item]) => (
+        <TableCell
+            align="center"
+            colSpan={isExpend(output_name) ? output_item.channels.length : 1}
+        >
+            {get(output_item, `source_id`) ? (
+                <Tooltip
+                    interactive
+                    title={get_source_tooltip(output_item)}
+                    placement="bottom"
+                    link
+                >
+                    <div>
+                        <ReferenceField
+                            record={output_item}
+                            basePath="/sources"
+                            label="Sources"
+                            source="source_id"
+                            reference="sources"
+                            link="show"
+                        >
+                            <ChipConditionalLabel source="label" />
+                        </ReferenceField>
+                    </div>
+                </Tooltip>
+            ) : null}
+        </TableCell>
+    ));
+
+const InputSourceAssociation = ({ isRowExpanded, input_item }) => (
+    <TableCell
+        align="center"
+        rowSpan={isRowExpanded ? input_item.channels.length + 1 : 1}
+    >
+        {input_item.parent.type === 'source' ? (
+            <ReferenceField
+                record={input_item}
+                basePath="/sources"
+                label="Sources"
+                source="parent.id"
+                reference="sources"
+                link="show"
+            >
+                <ChipConditionalLabel source="label" />
+            </ReferenceField>
+        ) : input_item.parent.type === 'receiver' ? (
+            <ReferenceField
+                record={input_item}
+                basePath="/receivers"
+                label="Receivers"
+                source="parent.id"
+                reference="receivers"
+                link="show"
+            >
+                <ChipConditionalLabel source="label" />
+            </ReferenceField>
+        ) : null}
```

Comment:
We discussed also whether blank cell was the right thing here. I wonder whether a simple text placeholder (instead of the chip) saying "No parent" might be nice?